core/sdk-core/src/main/java/software/amazon/awssdk/core/client/handler/BaseAsyncClientHandler.java

```diff
@@ -167,122 +173,4 @@ public void close() {
-        }
-
-        @Override
-        public void onError(Throwable error) {
-            delegate.onError(error);
-        }
-
-        @Override
-        public void onStream(Publisher<ByteBuffer> publisher) {
-            Optional<Publisher<ByteBuffer>> newPublisher = context.interceptorChain()
-                                                                  .modifyAsyncHttpResponse(context.interceptorContext()
-                                                                                                  .toBuilder()
-                                                                                                  .responsePublisher(publisher)
-                                                                                                  .build(),
-                                                                                 context.executionAttributes())
-                                                                  .responsePublisher();
-
-            if (newPublisher.isPresent()) {
-                delegate.onStream(newPublisher.get());
-            } else {
-                delegate.onStream(publisher);
-            }
-        }
-
-        @Override
-        public CompletableFuture<T> prepare() {
-            return delegate.prepare();
-        }
-    }
-
-    /**
-     * Adapter to {@link AsyncResponseTransformer} that performs unmarshalling and calls {@link
-     * software.amazon.awssdk.core.interceptor.ExecutionInterceptor}
-     * callbacks.
-     *
-     * @param <OutputT> Unmarshalled POJO response type.
-     * @param <ReturnT> Return type of {@link AsyncResponseTransformer}
-     */
-    private class UnmarshallingSdkHttpResponseHandler<OutputT extends SdkResponse, ReturnT>
-        implements TransformingAsyncResponseHandler<ReturnT> {
-
-        private final AsyncResponseTransformer<OutputT, ReturnT> asyncResponseTransformer;
-        private final ExecutionContext executionContext;
-        private final HttpResponseHandler<OutputT> responseHandler;
-        private CompletableFuture<ReturnT> transformFuture;
-
-        UnmarshallingSdkHttpResponseHandler(AsyncResponseTransformer<OutputT, ReturnT> asyncResponseTransformer,
-                                            ExecutionContext executionContext,
-                                            HttpResponseHandler<OutputT> responseHandler) {
-            this.asyncResponseTransformer = asyncResponseTransformer;
-            this.executionContext = executionContext;
-            this.responseHandler = responseHandler;
-        }
-
-        @Override
-        public void onHeaders(SdkHttpResponse response) {
-            try {
-                // TODO would be better to pass in AwsExecutionAttributes to the async response handler so we can
-                // provide them to HttpResponseHandler
-                OutputT resp =
-                    decorateResponseHandlers(responseHandler, executionContext)
```

Comment:
`decorateResponseHandler` is moved up to `BaseAsyncClientHandler#execute`

https://github.com/aws/aws-sdk-java-v2/pull/1077/files#diff-43d6f14f11805b1194455c311876e0a6R78